tests/test_ec2/test_elastic_network_interfaces.py

```diff
@@ -401,6 +401,43 @@ def test_elastic_network_interfaces_get_by_description():
     enis.should.have.length_of(0)
 
 
+@mock_ec2_deprecated
+@mock_ec2
+def test_elastic_network_interfaces_get_by_attachment_instance_id():
+    conn = boto.connect_vpc("the_key", "the_secret")
+    vpc = conn.create_vpc("10.0.0.0/16")
+    subnet = conn.create_subnet(vpc.id, "10.0.0.0/18")
+    ec2_resource = boto3.resource("ec2", region_name="us-east-1")
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+
+    security_group1 = conn.create_security_group(
+        "test security group #1", "this is a test security group"
+    )
+
+    create_instances_result = ec2_resource.create_instances(
+        ImageId="ami-d3adb33f", MinCount=1, MaxCount=1
```

Comment:
The warning pops up for quite a few tests - I think it's to do with the fact that this ImageId does not exist in our mocked world.
Don't worry this for now - we'll clean all those warnings up in one go (at some point..)